instance/models/mixins/ansible.py

```diff
@@ -20,98 +20,63 @@
 Instance app model mixins - Ansible
 """
 import os
+from collections import namedtuple
 from django.conf import settings
-from django.core.validators import MinValueValidator
 from django.db import models
-from django.template import loader
 
 from instance import ansible
 from instance.repo import open_repository
 from instance.utils import poll_streams
 
 
-class AnsibleInstanceMixin(models.Model):
+Playbook = namedtuple('Playbook', [
+    'source_repo',  # Path or URL to a git repository contianing the playbook to run
+    'playbook_filename',  # Relative path to the playbook within source_repo
+    'requirements_path',  # Relative path to a python requirements file to install before running the playbook
```

Comment:
@bradenmacdonald It would be good to make the naming consistent here, i.e. to use `playbook_filename`/`requirements_filename` or `playbook_path`/`requirements_path`. I'm leaning towards the latter option, which seems to be what you're using in `_run_playbook` below.